src/main/cpp/src/parse_uri.cu

```diff
@@ -494,48 +494,42 @@ bool __device__ validate_fragment(string_view fragment)
 
 __device__ std::pair<string_view, bool> find_query_part(string_view haystack, string_view needle)
 {
-  auto const n_bytes     = needle.size_bytes();
-  auto const find_length = haystack.size_bytes() - n_bytes + 1;
-
-  auto h           = haystack.data();
-  auto const end_h = haystack.data() + find_length;
-  auto n           = needle.data();
-  bool match       = false;
-  while (h < end_h) {
-    match = false;  // initialize to false to prevent empty query key
-    for (size_type jdx = 0; (jdx == 0 || match) && (jdx < n_bytes); ++jdx) {
-      match = (h[jdx] == n[jdx]);
+  auto const n_bytes = needle.size_bytes();
+  auto h             = haystack.data();
+  auto const h_end   = h + haystack.size_bytes();
+  auto n             = needle.data();
+
+  // stop matching early after it can no longer contain the string we are searching for
+  while (h + n_bytes < h_end) {
+    bool match_needle = false;  // initialize to false to prevent empty query key
```

Comment:
Seems can not handle the corner case:
needle = '' empty string.
Set match_needle = true?

Is this a valid case:  find_query_part('=value', '') ?

Reply:
Very nice catch, thanks.
The test code to generate excepted results in java test has a bug, it returns null for `'=value', ''` this case, while spark return value. So you are right, it should be true here.